src/pages/home/report/ReportActionItemFragment.js

```diff
@@ -97,7 +97,9 @@ const ReportActionItemFragment = (props) => {
             }
             const {html, text} = props.fragment;
 
-            if (props.fragment.isDeletedParentAction) {
+            // Threaded messages display a [Deleted message] note instead of being hidden altogether.
+            // When not offline we want to immediately display this message whilst the delete action is pending
+            if ((!props.network.isOffline && props.hasCommentThread && props.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) || props.fragment.isDeletedParentAction) {
```

Comment:
I couldn't think of why we needed this change, so I when back to what we had before and it tests fine. Do you agree?

If it's a deleted parent action then we render the deleted message, and we set it in the optimistic data so it applies right away. OfflineWithFeedback has already been modified to handle the pending state well.

Reply:
Without this change, the [Deleted message] note will only display once the API request completes, rather than immediately. It's easier to see the difference if you test on a slower network (e.g. Slow 3G on Chrome DevTools). 

This is consistent with how it works if you delete a normal message on a slow network, i.e. it immediately is hidden, rather than hidden once the API request completes.

Let me know if the comment needs tidying up - I tried to keep it from getting too verbose, but maybe it's too unclear.

Reply:
Oh ok. I didn't know about the `previousMessage`. I see what you mean now. If we clean up the comment then I'm happy with this.

Reply:
Maybe something like this.

```
// Threaded messages display "[Deleted message]" instead of being hidden altogether.
// While offline we display the previous message with a strikethrough style. Once online we want to immediately display the deleted message while the delete action is pending
```

Reply:
That sounds much clearer, thanks!